File: Herencia/ejercicio 2/Python/Televisor.py
from Electrodomestico import Electrodomestico
class Television(Electrodomestico):
    def __init__(self, precioBase, color, consumoEnergico, peso, resolucion, sintonizador_tdt):
        super().__init__(precioBase, color, consumoEnergico, peso)
        self.resolucion = resolucion
        self.sintonizador_tdt = sintonizador_tdt
    def precioFinal(self):
        precio = super().precioFinal()
        if self.resolucion > 40:
            precio *= 1.30
        if self.sintonizador_tdt:
            precio += 50
        return precio

File: Herencia/ejercicio 2/Java/Televisor.java
class Television extends Electrodomestico {
    public int resolucion;
    public boolean sintonizadorTDT;
    public Television(double precioBase, String color, char consumoEnergetico, double peso, int resolucion, boolean tdt) {
        super(precioBase, color, consumoEnergetico, peso);
        this.resolucion = resolucion;
        this.sintonizadorTDT = tdt;
    }
    @Override
    public double precioFinal() {
        double precio = super.precioFinal();
        if (resolucion > 40) precio *= 1.30;
        if (sintonizadorTDT) precio += 50;
        return precio;
    }
}